### DISCUSSION TITLE
Getting the current LOD from Camera

### DISCUSSION THREAD

**Discussion by dmartin-gdms** [pelicanmapping/osgearth#1840] [Support Q&A] [answered]
Hello,

I am currently attempting to get a list of tiles that are currently displayed at the current LOD, but I am unsure of how to figure out what the current LOD is. Currently, I am creating an extent based off of my cameras view and assuming a LOD of 15. 
I then get a list of tilekeys at that LOD and attempt to create images from those keys. If no images can be created, I try again with a lower LOD. 
```
  unsigned int lod = 15;

  std::vector<osgEarth::TileKey> tileKeyVec;
  earthProfile->getIntersectingTiles(extendedSelectionExtent, lod, tileKeyVec);

  std::vector<osgEarth::GeoImage> foundImages;

  while (foundImages.empty() && lod >= 0)
  {
    for (auto const& key : tileKeyVec)
    {
      auto hasImage = cts->createImageImplementation(key, nullptr);
      if (hasImage.valid())
      {
        foundImages.push_back(hasImage);
      }
    }

    if (foundImages.empty())
    {
      lod -= 1;
      tileKeyVec.clear();
      earthProfile->getIntersectingTiles
        (extendedSelectionExtent, lod, tileKeyVec);
    }
  }
```

This "works", but I feel like I could be doing something different -- is there a way to calculate what the current LOD of each layer is based off of the cameras range & viewport size?

**Comment by gwaldron** [comment, marked as answer]
Not really.
LOD is not selected on a per-layer basis. At cull time each tile can have a different LOD depending on the view matrix, the culling parameters, and the map projection. So there is never a single LOD that you are seeing for the whole terrain (unless you are in a projected map with an orthographic projection matrix).
